templates/cmd.txt

```diff
@@ -0,0 +1,161 @@
+{%- import "utils/batch.txt" as batch -%}
+
+{%- macro    cd(directory, tabs) -%} if /i not "%$p%CD%$p%"=="{{ directory }}" (^
+{%- call batch::indent(tabs) %}    %__builtin_cd% {{ directory|safe }}^
+{%- call batch::indent(tabs) %}    ^&^& set "OLDPWD=%$p%CD%$p%"^
+{%- if echo -%}
+{%- call batch::indent(tabs) %}    ^&^& {%~ call pwd(tabs + 1) -%}^
+{%- endif -%}
+{%- if hook != InitHook::None -%}
+{%- call batch::indent(tabs) %}    ^&^& for /f "delims=" %$p%a in ('"%__builtin_pwd%"') do @if /i not "%$p%~dpa"=="%$p%~fa" (^
+{%- call batch::indent(tabs) %}        zoxide add -- "%$p%~fa"^
+{%- call batch::indent(tabs) %}    ) else (^
+{%- call batch::indent(tabs) %}        zoxide add -- "%$p%~fa\"^
+{%- call batch::indent(tabs) %}    )^
+{%- endif -%}
+{%- call batch::indent(tabs) %})
+{%- endmacro cd -%}
+
+{%- macro    pwd(tabs) -%} (^
+{%- if resolve_symlinks -%}
+{%- call batch::indent(tabs) %}    (^
+{%- call batch::indent(tabs) %}        for /f "skip=9 tokens=1,2,*" %$p%j in ('"%$fsutil% reparsepoint query ."') do @(^
+{%- call batch::indent(tabs) %}            if "%$p%~j"=="Print" if "%$p%~k"=="Name:" if not "%$p%~l"=="" (echo(%$p%~l)^
+{%- call batch::indent(tabs) %}        )^
+{%- call batch::indent(tabs) %}    ) ^|^| %__builtin_pwd%^
+{%- else -%}
+{%- call batch::indent(tabs) %}    %__builtin_pwd%^
+{%- endif -%}
+{%- call batch::indent(tabs) %})
+{%- endmacro pwd -%}
+
+{%- let section = "@rem ==========================================================================\n@rem" -%}
+{%- let not_configured = "@rem -- not configured --" -%}
+
+@echo off & setlocal DisableDelayedExpansion EnableExtensions
```

Comment:
Is it possible to reset these to their original values after initialization is done? Or do we need it to be enabled throughout?

Reply:
If the contents of `zoxide init cmd` were to be run as a script, then `endlocal` take care of that. Command-echoing is per-file, so it will be reset naturally at EOF (or sooner if e.g. terminated via `exit`).

In other case (`zoxide init cmd | cmd ...`) the environment changes will be discarded and nothing will be propagated to the parent shell (except for DOSKEY macro definitions, which are per-executable and pertain for the whole duration of a session). `setlocal`/`endlocal` don't work outside of scripts anyway.